### PR TITLE
analysis/dwarf: extract register profiles in headers

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Just move the code to clean up `librz/analysis/dwarf_process.c` a bit

**Test plan**

CI is green

### CODE REVIEW

**Comment by wargio** [comment]
you should extract also the others.

**Comment by XVilka** [comment]
> you should extract also the others.

There are no `librz/analysis/arch/XXX` directories for those. In times, after RzIL support for those will be added.

**Review by wargio — APPROVED** [review]

File: librz/analysis/arch/sh/sh_dwarf_regnum_table.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2012-2020 houndthe <cgkajm@gmail.com>
+// SPDX-FileCopyrightText: 2024 Billow <billow.fun@gmail.com>
```
Comment (wargio) [inline librz/analysis/arch/sh/sh_dwarf_regnum_table.h]: Put rizin and me.

File: librz/analysis/arch/ppc/ppc_dwarf_regnum_table.h [hunk 2]
```diff
@@ -0,0 +1,195 @@
+// SPDX-FileCopyrightText: 2012-2020 houndthe <cgkajm@gmail.com>
+// SPDX-FileCopyrightText: 2024 Billow <billow.fun@gmail.com>
```
Comment (wargio) [inline librz/analysis/arch/ppc/ppc_dwarf_regnum_table.h]: add me under this.

File: librz/analysis/arch/sh/sh_dwarf_regnum_table.h (hunk 1 above)
Comment (wargio) [reply librz/analysis/arch/sh/sh_dwarf_regnum_table.h]:
```
// SPDX-FileCopyrightText: 2024 RizinOrg <info@rizin.re>
// SPDX-FileCopyrightText: 2024 deroad <wargio@libero.it>
```

**Review by b1llow — APPROVED** [review]

**Pull request merged**